implementations/rust/ockam/ockam_api/src/authenticator/common.rs

```diff
@@ -0,0 +1,91 @@
+use crate::authenticator::direct::{OCKAM_ROLE_ATTRIBUTE_ENROLLER_VALUE, OCKAM_ROLE_ATTRIBUTE_KEY};
+use crate::authenticator::{AuthorityMember, AuthorityMembersRepository};
+use minicbor::Decoder;
+use ockam::identity::Identifier;
+use ockam_core::api::{RequestHeader, Response};
+use ockam_core::{Result, Routed};
+use ockam_node::Context;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+/// Return a response on the return route stating that a secure channel is needed to access
+/// the service
+pub async fn secure_channel_required(c: &mut Context, m: Routed<Vec<u8>>) -> Result<()> {
+    // This was, actually, already checked by the access control. So if we reach this point
+    // it means there is a bug.  Also, if it' already checked, we should receive the Peer'
```

Comment:
s/it'/it's

Reply:
I think that the whole support for `secure_channel_required` should be reworked to:

 - Avoid duplication where it is used.
 - Avoid the double parsing or requests (by having better message types on the workers).

Reply:
Yeah, a lot of code was just moved, this is one of the examples. Improving all the pieces that I move would make me work on it for another month)
P.S. What do you mean by "double parsing" here?